// react-native-agora/lib/module/internal/MediaEngineInternal.js
import { createCheckers } from 'ts-interface-checker';
import { IMediaEngineImpl } from '../impl/IAgoraMediaEngineImpl';
import AgoraMediaBaseTI from '../ti/AgoraMediaBase-ti';
const checkers = createCheckers(AgoraMediaBaseTI);
import { DeviceEventEmitter, EVENT_TYPE } from './IrisApiEngine';
export class MediaEngineInternal extends IMediaEngineImpl {
  static _audio_frame_observers = [];
  static _video_frame_observers = [];
  static _video_encoded_frame_observers = [];
  registerAudioFrameObserver(observer) {
    if (!MediaEngineInternal._audio_frame_observers.find(value => value === observer)) {
      MediaEngineInternal._audio_frame_observers.push(observer);
    }
    return super.registerAudioFrameObserver(observer);
  }
  unregisterAudioFrameObserver(observer) {
    MediaEngineInternal._audio_frame_observers = MediaEngineInternal._audio_frame_observers.filter(value => value !== observer);
    return super.unregisterAudioFrameObserver(observer);
  }
  registerVideoFrameObserver(observer) {
    if (!MediaEngineInternal._video_frame_observers.find(value => value === observer)) {
      MediaEngineInternal._video_frame_observers.push(observer);
    }
    return super.registerVideoFrameObserver(observer);
  }
  unregisterVideoFrameObserver(observer) {
    MediaEngineInternal._video_frame_observers = MediaEngineInternal._video_frame_observers.filter(value => value !== observer);
    return super.unregisterVideoFrameObserver(observer);
  }
  registerVideoEncodedFrameObserver(observer) {
    if (!MediaEngineInternal._video_encoded_frame_observers.find(value => value === observer)) {
      MediaEngineInternal._video_encoded_frame_observers.push(observer);
    }
    return super.registerVideoEncodedFrameObserver(observer);
  }
  unregisterVideoEncodedFrameObserver(observer) {
    MediaEngineInternal._video_encoded_frame_observers = MediaEngineInternal._video_encoded_frame_observers.filter(value => value !== observer);
    return super.unregisterVideoEncodedFrameObserver(observer);
  }
  release() {
    MediaEngineInternal._audio_frame_observers = [];
    MediaEngineInternal._video_frame_observers = [];
    MediaEngineInternal._video_encoded_frame_observers = [];
    this.removeAllListeners();
    super.release();
  }
  _addListenerPreCheck(eventType) {
    var _checkers$IAudioFrame, _checkers$IVideoFrame, _checkers$IVideoEncod;
    if ((_checkers$IAudioFrame = checkers.IAudioFrameObserver) !== null && _checkers$IAudioFrame !== void 0 && _checkers$IAudioFrame.strictTest({
      [eventType]: undefined
    })) {
      if (MediaEngineInternal._audio_frame_observers.length === 0) {
        this.registerAudioFrameObserver({});
      }
    }
    if ((_checkers$IVideoFrame = checkers.IVideoFrameObserver) !== null && _checkers$IVideoFrame !== void 0 && _checkers$IVideoFrame.strictTest({
      [eventType]: undefined
    })) {
      if (MediaEngineInternal._video_frame_observers.length === 0) {
        this.registerVideoFrameObserver({});
      }
    }
    if ((_checkers$IVideoEncod = checkers.IVideoEncodedFrameObserver) !== null && _checkers$IVideoEncod !== void 0 && _checkers$IVideoEncod.strictTest({
      [eventType]: undefined
    })) {
      if (MediaEngineInternal._video_encoded_frame_observers.length === 0) {
        this.registerVideoEncodedFrameObserver({});
      }
    }
    return true;
  }
  addListener(eventType, listener) {
    this._addListenerPreCheck(eventType);
    const callback = (eventProcessor, data) => {
      if (eventProcessor.type(data) !== EVENT_TYPE.IMediaEngine) {
        return;
      }
      eventProcessor.func.map(it => {
        it({
          [eventType]: listener
        }, eventType, data);
      });
    };
    // @ts-ignore
    listener.agoraCallback = callback;
    DeviceEventEmitter.addListener(eventType, callback);
  }
  removeListener(eventType, listener) {
    DeviceEventEmitter.removeListener(eventType,
    // @ts-ignore
    (listener === null || listener === void 0 ? void 0 : listener.agoraCallback) ?? listener);
  }
  removeAllListeners(eventType) {
    DeviceEventEmitter.removeAllListeners(eventType);
  }
}
